.devcontainer/devcontainer.json

```
@@ -0,0 +1,40 @@
+{
+  "name": "community.general devcontainer",
+  "image": "mcr.microsoft.com/devcontainers/python:3.14-bookworm",
+  "features": {
+    "ghcr.io/devcontainers/features/docker-in-docker:2": {}
+  },
+  "customizations": {
+    "vscode": {
+      "settings": {
+        "terminal.integrated.shell.linux": "/bin/bash",
+        "python.pythonPath": "/usr/local/bin/python",
+        "editor.defaultFormatter": "charliermarsh.ruff",
+        "editor.formatOnSave": true,
+        "files.autoSave": "afterDelay",
+        "files.eol": "\n",
+        "files.insertFinalNewline": true,
+        "files.trimFinalNewlines": true,
+        "files.trimTrailingWhitespace": true
+      },
+      "extensions": [
+        "charliermarsh.ruff",
+        "chrislajoie.vscode-modelines",
+        "eamodio.gitlens",
+        "esbenp.prettier-vscode",
+        "michelemelluso.gitignore",
+        "ms-python.python",
+        "ms-python.vscode-pylance",
+        "oderwat.indent-rainbow",
+        "redhat.ansible",
+        "redhat.vscode-yaml",
+        "trond-snekvik.simple-rst",
+        "wayou.vscode-todo-highlight"
+      ]
+    }
+  },
```

Most of them are extensions I use and I just included them there as a convenience, but on second thought after you questioning, I think you're right, we should not force this upon the users. Therefore:
```suggestion
        "ms-python.python",
        "ms-python.vscode-pylance",
        "redhat.ansible",
        "redhat.vscode-yaml",
        "trond-snekvik.simple-rst",
      ]
    }
  },
```